parsers/nmparser/read_cov_test.go

```diff
@@ -0,0 +1,89 @@
+				" THETA6        0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" THETA7        0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" THETA8        0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" THETA9        0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" SIGMA(1,1)   -1.69174E-06  2.72226E-05 -2.66410E-05 -8.57217E-06  1.50750E-07  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  5.38834E-09 -1.35902E-07  0.00000E+00  2.47065E-09  0.00000E+00  0.00000E+00 -1.05448E-08",
+				" OMEGA(1,1)    3.87129E-04 -2.38849E-03 -5.20942E-04  7.63109E-04  6.32715E-06  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00 -1.35902E-07  9.20604E-05  0.00000E+00  3.55059E-06  0.00000E+00  0.00000E+00  1.58649E-06",
+				" OMEGA(2,1)    0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" OMEGA(2,2)   -2.69021E-04  2.19637E-03  2.75030E-04  7.27702E-05  2.54001E-06  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  2.47065E-09  3.55059E-06  0.00000E+00  1.24812E-05  0.00000E+00  0.00000E+00 -3.22735E-07",
+				" OMEGA(3,1)    0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" OMEGA(3,2)    0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00",
+				" OMEGA(3,3)    1.02250E-04 -2.16771E-04  1.78136E-04 -5.31316E-05  2.49581E-06  0.00000E+00  0.00000E+00  0.00000E+00  0.00000E+00 -1.05448E-08  1.58649E-06  0.00000E+00 -3.22735E-07  0.00000E+00  0.00000E+00  3.47626E-06",
+			},
+			expected: false,
+			context:  "no zero",
+		},
+	}
+
+	for _, tt := range tests {
+		extData := parseCovLines(tt.lines)
+		assert.Equal(t, tt.lines[0], extData.EstimationMethods[0], "Fail :"+tt.context)
+
+		res := GetThetaValues(tt.lines)
+		for _, fa := range res {
+			assert.Equal(t, -0.214341, fa.Values[9], "Fail :"+tt.context)
```

Comment:
is there a reason just to spot check 1 value? I would at least also check the dimensions to make sure it gets those correct. For example, this would not catch if it overconsumeed the elements needed and also grabbed omegas/simgas right?

Reply:
agreed, I will add more checks.